hera_pspec/uvpspec.py

```diff
@@ -2712,6 +2712,41 @@ def combine_uvpspec(uvps, merge_history=True, verbose=True):
     return u
 
 
+def recursive_combine_uvpspec(uvps):
+    """
+    Method for faster combination of UVPSpec objects by combining them recursively.
+    This is faster than combine_uvpspec for long lists of files---e.g. if you have 
+    one uvpspec object for every unique baseline and hundreds of baselines. Note:
+    Histories are not merged, so this is the equivalent of running combine_uvpspec 
+    with merge_history=False. 
+    
+    Parameters
+    ----------
+    uvps : list
+        A list of UVPSpec objects to combine.
+
+    Returns
+    -------
+    u : UVPSpec object
+        A UVPSpec object with the data of all the inputs combined.
+
+    """
+    if len(uvps) == 0:
```

Comment:
Could you add the same tests to `combine_uvpspec` or move them there? only `len(uvps)==0` case is covered in `combine_uvpspec` right now (through `get_uvp_overlap`)

Reply:
Sure